File: Algorithms/p187_Repeated_DNA_Sequences/p187_Repeated_DNA_Sequences_2.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        res = set()
        seen = set()
        for i in range(10,len(s)+1):
            s1 = s[i-10:i]
            if s1 in seen:
                res.add(s1)
            else:
                seen.add(s1)
        return list(res)

File: Algorithms/p187_Repeated_DNA_Sequences/p187_Repeated_DNA_Sequences_2.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        vector<string> res;
        unordered_map<string,int> cnt;
        for(int i=10;i<=s.size();++i){
            string s1(s,i-10,10);
            if(++cnt[s1]==2) res.push_back(s1);
        }
        return res;
    }
};
